mtp_api/apps/transaction/models.py

```diff
@@ -14,6 +14,7 @@
 
 
 class Transaction(TimeStampedModel):
+    # TODO, not sure we need this, we might decide to delete it
     upload_counter = models.PositiveIntegerField()
```

Comment:
I wonder if we should replace this with a list of hashes of uploaded transaction files so far, or something, so we can avoid double uploads? Totally separate issue, though…